Use GenServer to define stopwatch (#7)
dwyl/phoenix-liveview-stopwatch: README.md

  end

  def get_timer_state(server) do
    GenServer.call(server, :state)
  end

  def reset(server) do
    GenServer.call(server, :reset)
  end

  # Server
  @impl true
  def init(:ok) do
    {:ok, {:stopped, ~T[00:00:00]}}
  end

  @impl true
  def handle_call(:start, _from, {_status, time}) do
    Process.send_after(self(), :tick, 1000)
    {:reply, :running, {:running, time}}
  end

  @impl true
  def handle_call(:stop, _from, {_status, time}) do
    {:reply, :stopped, {:stopped, time}}
  end

  @impl true
  def handle_info(:tick, {status, time} = stopwatch) do
    if status == :running do
      Process.send_after(self(), :tick, 1000)
      notify()
      {:noreply, {status, Time.add(time, 1, :second)}}
    else
      {:noreply, stopwatch}
    end
  end

  @impl true
  def handle_call(:state, _from, stopwatch) do
    {:reply, stopwatch, stopwatch}
  end

  @impl true
  def handle_call(:reset, _from, _stopwatch) do
    {:reply, :reset, {:stopped, ~T[00:00:00]}}
  end

  def subscribe() do
    PubSub.subscribe(Stopwatch.PubSub, "liveview_stopwatch")
  end

  def notify() do
    PubSub.broadcast(Stopwatch.PubSub, "liveview_stopwatch", :timer_updated)
  end
end
```

> Copmare to Agent, GenServer split functions into client and server logic.

Reviewer: ```suggestion
Compared to `Agent`, `GenServer` splits functions into client and server logic.
```

We can define the same client api functions name and use `hand_call` to send
message to the GenServer to stop, start, reset the stopwatch.

The ticking process is now done by calling `Process.send_after(self(), :tick 1000)`.

> The GenServer will then manage the ticks with `handle_info(:tick, stopwatch)`.

Reviewer: ```suggestion
The `GenServer` will then manage the `tick` events with `handle_info(:tick, stopwatch)`.
```

Now that we have define our server we need to update `lib/stopwatch/application.ex` to use
the GenServer instead of the Agent:

```elixir
    children = [
      # Start the Telemetry supervisor
      StopwatchWeb.Telemetry,
      # Start the PubSub system
      {Phoenix.PubSub, name: Stopwatch.PubSub},
      # Start the Endpoint (http/https)
      StopwatchWeb.Endpoint,
      # Start a worker by calling: Stopwatch.Worker.start_link(arg)
      # {Stopwatch.Worker, arg}

>       # {Stopwatch.Timer, name: Stopwatch.Timer}

Reviewer: ```suggestion
```
